common/persistence/shard_manager.go

```diff
@@ -101,3 +101,10 @@ func (m *shardManagerImpl) UpdateShard(request *UpdateShardRequest) error {
 	}
 	return m.shardStore.UpdateShard(internalRequest)
 }
+
+func (m *shardManagerImpl) CloseShard(request *CloseShardRequest) error {
+	internalRequest := &InternalCloseShardRequest{
+		ShardID: request.ShardID,
+	}
+	return m.shardStore.CloseShard(internalRequest)
```

Comment:
Need to understand, decide, and document what are the concurrency semantics here.   Are you expecting this fn to block until the shard is *really* closed?  Or do we use this to trigger shard shutdown and allow it to proceed in the background?  Should we convey to the persistence layer whether we want an immediate shutdown or a graceful one?  Should that be a hint or a command?  Needs a bit of discussion (including @wxing1292  @yiminc )